run.py

```diff
@@ -904,89 +905,87 @@ def postprocess_all(settings):
     #################################
     #  RUN THE SIMULATION WORKFLOW  #
     #################################
-    for year in range(start_year, end_year, travel_model_freq):
-
+    state = WorkflowState.from_settings(settings)
+    for year in state:
         # 1. FORECAST LAND USE
-        if land_use_enabled:
-
+        if state.should_do(WorkflowState.Stage.land_use):
             # 1a. IF START YEAR, WARM START MANDATORY ACTIVITIES
-            if (year == start_year) and (warm_start_activities):
-
+            if (state.is_start_year()) and warm_start_activities:
                 # IF ATLAS ENABLED, UPDATE USIM INPUT H5
                 if vehicle_ownership_model_enabled:
                     run_atlas_auto(settings, year, client, warm_start_atlas=True)
                 warm_start_activities(settings, year, client)
 
             # 1b. RUN LAND USE SIMULATION
-            forecast_year = min(year + travel_model_freq, end_year)
-            forecast_land_use(settings, year, forecast_year, client, container_manager)
-
-        else:
-            forecast_year = start_year
+            forecast_land_use(settings, year, state.forecast_year, client, container_manager)
+            state.complete(WorkflowState.Stage.land_use)
 
         # 2. RUN ATLAS (HOUSEHOLD VEHICLE OWNERSHIP)
-        if vehicle_ownership_model_enabled:
-
+        if state.should_do(WorkflowState.Stage.vehicle_ownership_model):
             # If the forecast year is the same as the base year of this
             # iteration, then land use forecasting has not been run. In this
             # case, atlas need to update urbansim *inputs* before activitysim
             # reads it in the next step.
-            if forecast_year == year:
+            if state.forecast_year == year:
                 run_atlas_auto(settings, year, client, warm_start_atlas=True)
 
             # If urbansim has been called, ATLAS will read, run, and update
             # vehicle ownership info in urbansim *outputs* h5 datastore.
             else:
-                run_atlas_auto(settings, forecast_year, client, warm_start_atlas=False)
+                run_atlas_auto(settings, state.forecast_year, client, warm_start_atlas=False)
+            state.complete(WorkflowState.Stage.vehicle_ownership_model)
 
         # 3. GENERATE ACTIVITIES
-        if activity_demand_enabled:
+        if state.should_do(WorkflowState.Stage.activity_demand):
 
             # If the forecast year is the same as the base year of this
             # iteration, then land use forecasting has not been run. In this
             # case we have to read from the land use *inputs* because no
             # *outputs* have been generated yet. This is usually only the case
             # for generating "warm start" skims, so we treat it the same even
             # if the "warm_start_skims" setting was not set to True at runtime
-            if forecast_year == year:
```

Comment:
We don't need to handle the default scenario anymore?

Reply:
The original scenario is quite complex though it can be reduced to the one that is presented in these changes. You can see later in the code assigning the parameter:
`warm_start=warm_start_skims or not land_use_enabled`